src/mococrw/key.h

```diff
@@ -40,24 +40,13 @@ class AsymmetricKey
     {
     }
 
-/*
- * If we support more than RSA keys at some point, this will become necessary
- * When that happens, we will also need to expose these paramters via the Spec class
- */
-#if 0
-    enum class Types : int { RSA = EVP_PKEY_RSA, };
-
-    Types getType() const  { return reinterpret_cast<Types>(EVP_PKEY_type(_key.get()->type)); }
-private:
-    //note that this unique_ptr will not compile because of the incompleteness of type Spec, but
-    //this is just to hint at what is to come
-    std::unique_ptr<Spec> _spec = nullptr;
-
-    //this we then want to call from the constructor
-    void _createSpecFromPKEY() { switch(getType()) .... }
-
-#endif
+    /**
+     * Supported asymmetric key types.
+     */
+    enum class KeyTypes : int { RSA = EVP_PKEY_RSA, ECC = EVP_PKEY_EC };
 
+    KeyTypes getType() const  { return static_cast<KeyTypes>(openssl::_EVP_PKEY_type(_key.get())); }
+    openssl::ellipticCurveNid getCurve() const;
```

Comment:
I have a bad gut feeling about methods that only work sometimes depending on the state of the object. I would like to centralize this somehow. Could we have a method instead to get the KeySpec that is then the respective keyspec depending on the type of key and contains all the context-dependendant information? 

I have the feeling that this is a cleaner general interface for the future.